netsim/augment/links.py

```diff
@@ -805,7 +805,12 @@ def set_interface_name(ifdata: Box, link: Box, ifcnt: int) -> None:
     ifdata.name = link.name
     return
 
-  node_name = link.interfaces[ifcnt].node
+  link_iface = link.interfaces[ifcnt]
```

Comment:
* The intent was absolutely spot-on
* The logic was incorrect -- interface **name** should take precedence over link **name**
* We already have a function that fetches an attribute from a hierarch of objects. Replaced the explicit logic with the call to that function.

Reply:
Aha, right. Thanks for this!